lib/build-manager.js

```diff
@@ -236,6 +245,148 @@ class BuildManager {
+  * @param {string} untarPath - path to untar
+  * @returns {function} function to untar
+  */
+  untarPackage(contentData, untarPath) {
+    const stream = through();
+    return function (callback) {
+      stream
+        .pipe(zlib.Unzip()) // eslint-disable-line new-cap
+        .pipe(tar.extract(untarPath))
+        .once('finish', () => {
+          callback();
+        });
+      stream.end(Buffer.from(contentData, 'base64'));
+    };
+  }
+
+  /**
+  *
+  * writes a list of base64 encoded assets to path
+  * @function writeToPath
+  * @param {object} builtAssets - { [fileName]: { data: [base64 endcoded content] } }
+  * @param {string} unpackPath - path to unpack
+  * @returns {function} function to unpack
+  */
+  writeToPath(builtAssets, unpackPath) {
+    return function (cb) {
+      async.each(Object.entries(builtAssets), function (entry, callback) {
+        const [fileName, fileContent] = entry;
+        const fileData = fileContent.data;
+        const writePath = unpackPath + '/' + fileName;
+        const buffer = Buffer.from(fileData, 'base64');
+        fs.mkdir(unpackPath, { recursive: true }, function (err) {
+          if (err) {
+            callback(err, null); // eslint-disable-line callback-return
+          } else {
+            fs.writeFile(writePath, buffer, 'base64', (error) => {
+              callback(error, null);
+            });
+          }
+        });
+      }, function (err, results) {
+        cb(err, results);
+      });
+    };
+  }
+  /**
+  *
+  * Walks directory and generate a list of file infos to pass
+  * to bffs.publish for all non-compressed files
+  * @function walkAndGenerateFileInfos
+  * @param {string} directory - directory
+  * @param {array} files - array of files to add to
+  * @returns {function} function to walk and generate
+  */
+  walkAndGenerateFileInfos(directory, files) {
+    return function (callback) {
+      const walker = walk(directory);
+      walker.on('errors', function (root, nodeStatsArray, cb) {
+        this.log.error('Error walking directory', nodeStatsArray);
+        cb();
+      });
+      walker.on('file', function found(root, file, cb) {
+        const filepath = path.join(root, file.name);
+        const extension = path.extname(filepath);
+        if (extension !== '.gz') {
+          const compressedPath = filepath + '.gz';
+          fs.exists(compressedPath, (exists) => {
+            const compressed = exists ? compressedPath : null;
+            fs.readFile(filepath, (err, fileContent) => {
+              if (err) throw err;
+              const fileInfo = {
+                content: filepath,
+                compressed,
+                fingerprint: fingerprinter(filepath, { content: fileContent }).id,
+                filename: file.name,
+                extension
+              };
+              files.push(fileInfo);
+              return cb();
+            });
+          });
+        } else {
+          return cb();
```

Comment:
Suggestion, return early by checking `extension === '.gz'` at the `if` block, will allow for removal of a set of indents.